lib/internal/errors.js

```diff
@@ -1095,6 +1095,12 @@ E('ERR_INVALID_PACKAGE_TARGET', (pkgPath, key, subpath, target) => {
   if (key === '.') {
     return `Invalid "exports" main target ${JSONStringify(target)} defined ` +
       `in the package config ${pkgPath}${sep}package.json`;
```

Comment:
It would be good to ensure the dot main also gets this message:

```suggestion
    const relError = typeof target === 'string' && target.length && !target.startsWith('./');
    return `Invalid "exports" main target ${JSONStringify(target)} defined ` +
      `in the package config ${pkgPath}${sep}package.json${relError ?
      ' - targets must start with "./"' : ''}`;
```

Using the same approach below might save a few lines too.

I also usually try to avoid adding a `.` after a file path as this disables link detection in the terminal in vscode.